src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -0,0 +1,231 @@
+            .build()
+    }
+
+    private fun authorization(vararg payloads: String): String {
+        val md = MessageDigest.getInstance("MD5")
+        val bytes = payloads.joinToString("").toByteArray()
+        val digest = md.digest(bytes)
+        return digest
+            .fold("") { str, byte -> str + "%02x".format(byte) }
+            .padStart(32, '0')
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapters = mutableListOf<SChapter>()
+        var currentPage = 0
+        do {
+            val chaptersDto = fetchChapterListPageable(manga, currentPage)
+            chapters += chaptersDto.toModel(langOption)
+            currentPage++
+        } while (chaptersDto.hasNextPage())
+        return Observable.from(listOf(chapters.reversed()))
+    }
+
+    private fun fetchChapterListPageable(manga: SManga, page: Int): ChapterPageDto {
+        return try {
+            val maxResult = 16
+            val pathSearch = when (langOption.lang) {
+                "it" -> langOption.infix
+                else -> "v3/po"
+            }
+            val url = "$apiUrl/api/$pathSearch/GetChapterListFilter/${manga.slug()}/$maxResult/$page/all/ASC"
+            return client
+                .newCall(GET(url, apiHeaders(url)))
+                .execute()
+                .toChapterPageDto()
+        } catch (e: Exception) {
+            Log.e("::fetchChapter", e.toString())
+            ChapterPageDto()
+        }
+    }
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun chapterListSelector() = ""
+
+    override fun imageUrlParse(document: Document) = ""
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val dto = response.htmlDocumentToDto()
+        return MangasPage(
+            mangas = dto.toLatestUpdatesModel(),
+            hasNextPage = dto.hasNextPageToLatestUpdates(),
+        )
+    }
+
+    override fun latestUpdatesFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun latestUpdatesNextPageSelector() = "#next-prev a:nth-child(2):not(.line-through)"
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/${langOption.infix}/latest-releases".toHttpUrl().newBuilder()
+            .addQueryParameter("page", "$page")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesSelector() = "main > div > table > tbody > tr"
+
+    override fun mangaDetailsParse(response: Response) =
+        response.htmlDocumentToDto().toMangaDetailsModel()
+
+    override fun mangaDetailsParse(document: Document) = throw UnsupportedOperationException()
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pageListData = document.htmlDocumentToDto().props.pageProps.pageListData
+        val decodedData = CryptoAES.decrypt(pageListData!!, "ABC@123#245")
```

Comment:
can the password be extracted from the page or a script? so as to not hardcode it

Reply:
I didn't find this password when I analyzed the JS code. 

@bapeey, could you tell about it?

Reply:
Key is in a static chunk in the reader html, maybe find the script with `script[src^=/_next/static/chunks/pages/%5Btype%5D/%5Bidmanga%5D]` and make a call